sql/RFC_SQL.md

```diff
@@ -0,0 +1,1165 @@
+the `release` method.
+
+```ts
+await client.release();
+```
+
+When the pool is no longer needed, it must be closed by the `close` method. The
+`close` method will close all the connections in the pool.
+
+```ts
+await client.close();
+```
+
+Using
+[Explicit Resource Management](https://github.com/tc39/proposal-explicit-resource-management),
+no manual release or close is needed.
+
+```ts
+await using pool = new ClientPool(connectionUrl, connectionOptions);
+await pool.connect();
+await using client = await pool.acquire();
+// no need to release the client at the end
+// no need to close the pool at the end
+```
+
+#### Querying the database
+
+The client provides the following methods for querying.
+
+- `execute`: Executes a SQL statement
+- `query`: Queries the database and returns an array of object
+- `queryOne`: Queries the database and returns at most one entry as an object
+- `queryMany`: Queries the database with an async generator and yields each
+  entry as an object. This is good for when you want to iterate over a massive
+  amount of rows.
+- `queryArray`: Queries the database and returns an array of arrays
+- `queryOneArray`: Queries the database and returns at most one entry as an
+  array
+- `queryManyArray`: Queries the database with an async generator and yields each
+  entry as an array. This is good for when you want to iterate over a massive
+  amount of rows.
+- `sql`: Allows you to create a query using template literals, and returns the
+  entries as an array of objects. This is a wrapper around `query`
+- `sqlArray`: Allows you to create a query using template literals, and returns
+  the entries as an array of arrays. This is a wrapper around `queryArray`
+
+See the [examples](#examples) section for sample usage.
+
+#### Prepared statement
+
+A prepared statement can also be created with the provided method.
+
+- `prepare`: Returns a `PreparedStatement` class
+
+The `PreparedStatement` class provides a subset of the client methods for
+querying.
+
+- `execute`: Executes a SQL statement
+- `query`: Queries the database and returns an array of object
+- `queryOne`: Queries the database and returns at most one entry as an object
+- `queryMany`: Queries the database with an async generator and yields each
+  entry as an object. This is good for when you want to iterate over a massive
+  amount of rows.
+- `queryArray`: Queries the database and returns an array of arrays
+- `queryOneArray`: Queries the database and returns at most one entry as an
+  array
+- `queryManyArray`: Queries the database with an async generator and yields each
+  entry as an array. This is good for when you want to iterate over a massive
+  amount of rows.
+
+See the [examples](#examples) section for sample usage.
+
+#### Transaction
+
+Transactions are also supported by using the provided methods
+
+- `beginTransaction`: Returns a `Transaction` class that implements the
```

Comment:
Yeah, I was researching the differences between databases, and since sqlite and pg uses `begin` I opted for that 💪🏻 My general rule of thumb for wording was to compare the three main ones (pg, mysql and sqlite) and go with the majority